compiler/rustc_parse/src/parser/item.rs

```diff
@@ -69,6 +69,21 @@ impl<'a> Parser<'a> {
                 let msg = &format!("expected item, found {}", token_str);
                 let mut err = self.struct_span_err(self.token.span, msg);
                 err.span_label(self.token.span, "expected item");
+                // Parses incorrect visibility order such as `async pub`.
+                if self.token == TokenKind::Ident(kw::Async, false) {
```

Comment:
I am not sure why I can't use the same method for `kw::Const` for `const pub` but weird that `self.token` is ` ` whitespace of an empty space.

Reply:
Wait, what? `self.token` should never have whitespace. What enum variant is `token.kind`?

Reply:
Weird, it was indeed `const`, `self.token=Token { kind: Ident("const", false), span: test3.rs:1:1: 1:6 (#0) }` but I don't know why it wasn't matched with `matches!(self.token.kind, TokenKind::Ident(kw::Async | kw::Const), false) {`.